rocketpy/EnvironmentAnalysis.py

```diff
@@ -605,12 +606,84 @@ def calculate_average_wind_profile(self):
         """average, 1, 2, 3 sigma wind profile from 0 35,000 ft AGL"""
         ...
 
-    # TODO: Implement
-    def calculate_average_day_wind_rose(self):
-        """average day wind rose"""
-        ...
+    def animate_average_wind_rose(self, figsize=(8, 8)):
+        """Animates the wind_rose of an average day. The inputs of a wind_rose are the location of the
+        place where we want to analyse, (x,y,z). The data is ensembled by hour, which means, the windrose
+        of a specific hour is generated by bringing together the data of all of the days available for that
+        specific hour. It's possible to change the size of the gif using the parameter figsize, which is the
+        height and width in inches.
+
+        Parameters
+        ----------
+        figsize : array
+
+        Returns
+        -------
+        Image : ipywidgets.widgets.widget_media.Image
+        """
+        days = list(self.surfaceDataDict.keys())
+        hours = list(self.surfaceDataDict[days[0]].keys())
+
+        windSpeed = {}
+        windDir = {}
+
+        for hour in hours:
+            windSpeed[hour] = []
+            windDir[hour] = []
+            for day in days:
+                windSpeed[hour].append(
+                    self.pressureLevelDataDict[day][hour]["windSpeed"](self.elevation)
+                )
+                windDir[hour].append(
+                    self.pressureLevelDataDict[day][hour]["windDirection"](
+                        self.elevation
+                    )
+                )
+
+        fig = plt.figure(
+            facecolor="w", edgecolor="w", figsize=figsize
+        )  # initialise la figure
+
+        filename = "wind_rose.gif"
+        FFMpegWriter = matplotlib.animation.writers["ffmpeg"]
+        metadata = dict(
+            title="windrose",
+            artist="windrose",
+            comment="""Made with windrose
+                http://www.github.com/scls19fr/windrose""",
+        )
+        writer = FFMpegWriter(fps=1, metadata=metadata)
+
+        with writer.saving(fig, filename, 100):
+            for hour in hours:
+                ax = WindroseAxes.from_ax(fig=fig)
+                ax.bar(
+                    windDir[hour],
+                    windSpeed[hour],
+                    normed=True,
+                    opening=0.8,
+                    edgecolor="white",
+                )
+                ax.set_title(
+                    f"Windrose of an average day. Hour {float(hour):05.2f}".replace(
+                        ".", ":"
+                    )
+                )
+
+                writer.grab_frame()
+                plt.clf()
+
+        with open(filename, "rb") as file:
+            image = file.read()
+
+        fig_width, fig_height = plt.gcf().get_size_inches() * fig.dpi
+        return widgets.Image(
+            value=image,
+            format="gif",
+            width=fig_width,
+            height=fig_height,
+        )
```

Comment:
I was not familiar with this way of generating an animation. Without running this on a jupyter notebook (i.e. running it as a stand alone script) would it still work or does using `widgets` mean that this only works inside a notebook?

Reply:
It generates a .gif, I can explicit that behavior in the documentation.
On the notebook: the gif will be displayed automatically
On the terminal: the gif is generated and the user can just open it

Reply:
I mean, it is not the best, but if I have the time I search a better optional, unless you think this would be a not functional behavior, in that case I would search how to display on both cases